jlm/llvm/opt/alias-analyses/RegionAwareMemoryNodeProvider.cpp

```diff
@@ -462,6 +464,48 @@ class RegionAwareMemoryNodeProvisioning final : public MemoryNodeProvisioning
   }
 
 private:
+  [[nodiscard]] bool
+  GetOutputNodesFromRegisterNode(
+      const rvsdg::output & output,
+      util::HashSet<const PointsToGraph::MemoryNode *> & memoryNodes) const
+  {
+    const PointsToGraph::RegisterNode * registerNode;
+    try
+    {
+      registerNode = &PointsToGraph_.GetRegisterNode(output);
+    }
+    catch (...)
+    {
+      return false;
+    }
+
+    for (auto & memoryNode : registerNode->Targets())
+      memoryNodes.Insert(&memoryNode);
+
+    return true;
+  }
+
+  [[nodiscard]] bool
+  GetOutputNodesFromRegisterSetNode(
```

Comment:
I cannot push it to the superclass (MemoryNodeProvisioning) as these methods require the points-to graph, but that one is not present in the superclass. I also do not want to extend the superclass with the points-to graph. The MemoryNodeProvisioning was more thought of as a pure interface and not as something that should hold data. Since it is temporary, do you think we can leave it as it is right now, since it will go away again anyway?

Reply:
Yes, I assumed there was a reason, just wanted to have it confirmed